### PR TITLE
LPS-88934 Support multiple entity selection

### PR DESCRIPTION
**Pull request by thektan**
http://issues.liferay.com/browse/LPS-88934

WIP, but selecting multiple organizations does create multiple rows. I still need to get it to pre-select the currently selected value in the list after closing and opening. Also I'm still using `Liferay.Util.selectEntity` but I need to switch it to use `A.LiferayItemSelectorDialog` since it does the same behavior for the multiple selection.

This is also rebased off of https://github.com/thektan/liferay-portal/pull/23.

cc @epgarcia The backend changes for organization are good so far! I don't see any issues with it at the moment.

### CODE REVIEW

**Comment by epgarcia** [comment]
Thanks @thektan! I will add it for userId on top of these changes.

**Comment by epgarcia** [comment]
@andresfulla please review Kevin's changes but DO NOT SEND the backend changes. I'll send them to Brian myself, after some refactor. I'll CC you and Kevin. Thanks

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js [hunk 1]
```diff
@@ -224,15 +224,40 @@ class CriteriaRow extends Component {
...
-		const {criterion, onChange} = this.props;
-
-		onChange(
-			{
-				...criterion,
-				type,
-				value
-			}
-		);
+		const {criterion, index, onAdd, onChange} = this.props;
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js]: I'd rather have the oData parser taking this decision.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/TypedInput.es.js [hunk 2]
```diff
@@ -97,24 +100,89 @@ class TypedInput extends React.Component {
...
+			value
 		} = this.props;
 
-		Liferay.Util.selectEntity(
-			{
-				dialog: {
-					constrain: true,
-					destroyOnHide: true,
-					modal: true
+		if (multiple) {
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/TypedInput.es.js]: Fantastic. Although, it'd be great to start extracting this big chucks of functionality into functions. Specially the ones related to out-of-react stuff. This simple component is getting to 300 lines. Just a suggestion.

**Review by andresfulla — COMMENTED** [review]
Just suggesting a couple of things.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js (hunk 1 above)
Comment (thektan) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js]: The most ideal way of displaying the selected items right now would be a new group reading something like `organization eq 'a' OR organization eq 'b'` and I thought it would be cleaner having data formatted in the same way sending it up to the odata parser.

I suppose it does make sense for this to be in the odata parser since this is a translation that we'll be doing converting an array to multiple rows.

I'll try handling this in the odata parser though. This could help expose the "custom" behavior and make it more ideal if we wanted to support `any` and `all` operators.

Comment (thektan) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js]: I've come to the conclusion that having the check for array is better to be done in `CriteriaRow` because not only would we have to update the odata parser to handle arrays, but the criteria builder as well to keep the `criteriaMap` in sync with the query. It seems like more overhead managing the cases in multiple areas than directly creating new rows where the `criteriaMap` and `query` change consistently.

TLDR: I'm keeping the changes already made in this PR for the array handling.

**Comment by thektan** [comment]
@andresfulla This is good to fully look over now! 

**My changes:** 757e1d4...97d9673

**Notable changes:**
- Created separate components for each of the possible rendered in `TypedInput`. The conditional for which input renders is now in `CriteriaRow` https://github.com/andresfulla/liferay-portal/pull/19/commits/f4d10c53d8d0228194051143083142b5dc8b64ba#diff-58cd48028a8547740567dd061ab71e28R268.
- Updated the Language.properties path now that they are located elsewhere. b3914bd

**Comment by andresfulla** [comment]
Thanks @thektan!! Changes look great. Reusability for the win!!

**Comment by andresfulla** [comment]
Just sent it to Bryce https://github.com/bryceosterhaus/liferay-portal/pull/24
